### ISSUE TITLE
typeof typeof (variable) yields a String anyway

### ISSUE THREAD

**Issue by prgrmmraben** [eslint/eslint#11246]
**The rule should do:**
Warn about repeated `typeof` as it always yield a String
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Description
]
Not paying attention, this resulted a bug on my project, and this could happen to any JavaScript developer.

**What category of rule is this?**
I don't really know as I am not aware of rules classes and also a n00b

```js
undefined undefined 0 // : string
undefined undefined Null // : string
undefined undefined infinity // : string
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Not really sure.

**Are you willing to submit a pull request to implement this rule?**
No
